components/board/pi/impl/i2c.go

```diff
@@ -88,6 +88,16 @@ func (s *piPigpioI2CHandle) WriteWordData(ctx context.Context, register byte, da
 	return nil
 }
 
+func (s *piPigpioI2CHandle) ReadBlockData(ctx context.Context, register byte, numBytes uint8) ([]byte, error) {
+	data := make([]byte, numBytes)
```

Comment:
should break read > 32 bytes into multiple read (per i2cReadBlockData documentation)

Reply:
Those reads would not be atomic, resulting in potentially inconsistent data. I'd rather return an error if you want to read more than 32 bytes. but I do like the suggestion of taking that 32-byte limit into account!